src/containers/BuildUnsignedConsolidation/GenericEcdsaForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { Form, FormikHelpers, FormikProvider, useFormik } from 'formik';
+import { Button, FormikTextfield } from '~/components';
+import { Link } from 'react-router-dom';
+import * as Yup from 'yup';
+
+const validationSchema = Yup.object({
+  userKey: Yup.string(),
+  backupKey: Yup.string(),
+  bitgoKey: Yup.string().required(),
+  walletPassphrase: Yup.string(),
+  startingScanIndex: Yup.number(),
+  endingScanIndex: Yup.number().moreThan(
+    Yup.ref('startingScanIndex'),
+    'Ending scan index must be greater than starting scan index'
+  ),
+  seed: Yup.string(),
+}).required();
+
+export type EcdsaFormValues =  Yup.Asserts<typeof validationSchema>;
+
+export type AdaFormProps = {
+  onSubmit: (
+    values: EcdsaFormValues,
+    formikHelpers: FormikHelpers<EcdsaFormValues>
+  ) => void | Promise<void>;
+}
+
+export function GenericEcdsaForm({ onSubmit }: AdaFormProps) {
+
+  const formik = useFormik<EcdsaFormValues>({
+    onSubmit,
+    initialValues: {
+      userKey: '',
+      backupKey: '',
+      bitgoKey: '',
+      walletPassphrase: '',
+      startingScanIndex: 1,
+      endingScanIndex: 21,
+      seed: '',
```

Comment:
an empty string is still considered a valid seed, to make sure this is correct:
```suggestion
      seed: undefined,
```